### PR TITLE
Provide more default values for proximity calculations

### PR DESCRIPTION
**Pull request by jzwar**
# Overview
Simplify syntax according to #163 

## Showcase
```
import splinepy
spline = splinepy.Bezier(
    degrees=[2, 1],
    control_points=[[0, 0], [1, 1], [2, 0], [0, 2], [1, 2], [2, 2]],
)
para_cords = spline.proximities(np.random.rand(10, 2))
```

## Checklists
* [ ] Documentations are up-to-date.
* [ ] Added example(s)
* [ ] Added test(s)

### CODE REVIEW

File: splinepy/spline.py [hunk 1]
```diff
@@ -1509,32 +1510,52 @@ def proximities(
...
+            return super().proximities(
+                queries=queries,
+                initial_guess_sample_resolutions=_default_if_none(
+                    tolerance, self.control_mesh_resolutions * 2
+                ),
+                tolerance=_default_if_none(tolerance, settings.TOLERANCE),
+                max_iterations=max_iterations,
+                aggressive_search_bounds=aggressive_search_bounds,
+                nthreads=_default_if_none(nthreads, settings.NTHREADS),
+            )[0]
```
Comment (j042) [inline splinepy/spline.py]:
```suggestion
        verbose_info = super().proximities(
                queries=queries,
                initial_guess_sample_resolutions=_default_if_none(
                    initial_guess_sample_resolutions, self.control_mesh_resolutions * 2
                ),
                tolerance=_default_if_none(tolerance, settings.TOLERANCE),
                max_iterations=max_iterations,
                aggressive_search_bounds=aggressive_search_bounds,
                nthreads=_default_if_none(nthreads, settings.NTHREADS),
        )

        if return_verbose:
            return verbose_info
        else:
            return verbose_info[0]
```
maybe non-verbose output can check the distance or convergence_norm against given tolerance and add warning log

Comment (jzwar) [reply splinepy/spline.py]: Good idea. I'll adjust it

Comment (jzwar) [reply splinepy/spline.py]: I updated the code according to your suggestions

**Review by j042 — APPROVED** [review]
Awesome, thanks!

**Pull request merged**